support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -47,42 +46,43 @@ class CreateZuoraSubscription(servicesProvider: ServiceProvider = ServiceProvide
     context: Context,
     services: Services
   ): FutureHandlerResult = {
-    val now = () => DateTime.now(DateTimeZone.UTC)
-    val today = () => LocalDate.now(DateTimeZone.UTC)
-    val promotionService = services.promotionService
-    val redemptionService = services.redemptionService
-    val zuoraGiftService = services.zuoraGiftService
-    val zuoraService = services.zuoraService
-    val giftCodeGenerator = services.giftCodeGenerator
-    val isTestUser = state.user.isTestUser
-    val config: ZuoraConfig = services.config.zuoraConfigProvider.get(isTestUser)
+
+    val createSubscription = {
```

Comment:
What is the thinking behind declaring these vals within a block?

Reply:
scoping - I realised there were a load of vals that were only used in the one place so this makes that clear.
It would defintely be more consistent to actually turn it into a separate private function, but that seemed overkill when I actually tried it out.
What you you reckon?

Reply:
I had a brain wave - see below https://github.com/guardian/support-frontend/pull/2756/files/d4271139f3be9cddb0bff2d5bfdc2e52077c5051#r501782946